src/components/SchemaTree.tsx

```diff
@@ -17,24 +20,55 @@ export interface ISchemaTree {
   expanded?: boolean;
   maxRows?: number;
   onGoToRef?: GoToRefHandler;
+  maskControlsHandler?: (attrs: Array<{ path: string; required: boolean }>) => string[];
+  updateMaskProp?: () => ReactElement;
+  maskProps?: Array<{ path: string; required: boolean }>;
 }
 
 const canDrag = () => false;
 
 export const SchemaTree = observer<ISchemaTree>(props => {
   const { hideTopBar, name, treeStore, maxRows, className, onGoToRef } = props;
 
-  treeStore.on(TreeListEvents.NodeClick, (e, node) => treeStore.toggleExpand(node));
-
   const itemData = {
     treeStore,
     count: treeStore.nodes.length,
     onGoToRef,
   };
 
+  const [selectedProps, setSelectedProps] = useState((props.maskProps || []) as Array<{
+    path: string;
+    required: any;
+  }>);
+
   const rowRenderer = React.useCallback(
-    (node, rowOptions) => <SchemaRow node={node} rowOptions={rowOptions} {...itemData} />,
-    [itemData.count],
+    (node, rowOptions) => {
+      const possibleProps = props.maskControlsHandler
```

Comment:
Suggest passing custom `rowRenderer` rather than having this here.
You can refer to https://github.com/stoplightio/tree-list/blob/master/src/components/TreeListItem.tsx#L39 to have a better understanding.